File: Data Structures & Algorithms/kth-smallest-integer-in-bst/submission-0.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    public int kthSmallest(TreeNode root, int k) {
        int[] count = new int[1];
        count[0] = 0;
        return helper(root, k, count).val;   
    }

    private TreeNode helper(TreeNode node, int k, int[] count) {
        if (node == null) {
            return null;
        } else {
            TreeNode left = helper(node.left, k , count);
            if (left != null) {
                return left;
            }
            count[0]++;
            if (count[0] == k) {
                return node;
            }
            TreeNode right = helper(node.right, k, count);
            if (right != null) {
                return right;
            }
            return null;
        }
    }
}

File: Data Structures & Algorithms/palindromic-substrings/submission-0.py
class Solution:
    def countSubstrings(self, s: str) -> int:
        count = 0
        left = 0
        while left < len(s):
            right = left + 1
            tempCount = 1
            while right < len(s) and s[left] == s[right]:
                tempCount += 1        
                right += 1
            tempCount = (tempCount * (tempCount + 1)) // 2
            count += tempCount
            temp = right
            while left > 0 and right < len(s) and s[left - 1] == s[right]:
                count += 1
                left -= 1
                right += 1
            left = temp
        return count